### DISCUSSION TITLE
How to render MermaidJS code in an HTML file?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [mermaid-js/mermaid#5977] [Q&A] [answered]
Rendering it in VS Code in a `.Markdown` file is trivial:

~~~MD
```mermaid
```
~~~

However, how to render it in an `.HTML` file is not so. The sole question online asking this that I've been able to locate thus far is [`stackoverflow.com/questions/74086784`](https://stackoverflow.com/questions/74086784/how-to-embed-mermaid-in-html-file#comment139479486_74086784), whose sole response is nondescript.

**Comment by roke-julian-lockhart** [comment]
> [`#discussion-7336721`](https://github.com/orgs/mermaid-js/discussions/5977#discussion-7336721)

Per https://github.com/mermaid-js/mermaid-cli/issues/48#issuecomment-687015287, [`getting-started.md#requirements-for-the-mermaid-api`](https://github.com/mermaid-js/mermaid/blob/c7fa906115328783abeb9b6b08075fc39d94bfca/packages/mermaid/src/docs/intro/getting-started.md#requirements-for-the-mermaid-api:~:text=Requirements%20for%20the%20Mermaid%20API) might be suitable documentation. I'm trying to render this in a single HTML *document*, not a web page, ~~so I'm not confident it'll work (without a server)~~.

**Reply by roke-julian-lockhart** [reply, marked as answer]
> [`#discussioncomment-10982775`](https://github.com/orgs/mermaid-js/discussions/5977#discussioncomment-10982775)

Indeed, [`getting-started.md#requirements-for-the-mermaid-api`](https://github.com/mermaid-js/mermaid/blob/c7fa906115328783abeb9b6b08075fc39d94bfca/packages/mermaid/src/docs/intro/getting-started.md#requirements-for-the-mermaid-api:~:text=Requirements%20for%20the%20Mermaid%20API) worked, for:

<blockQuote>

```HTML
<html>
  <body>
    Here is one mermaid diagram:
    <pre class="mermaid">
            graph TD
            A[Client] --> B[Load Balancer]
            B --> C[Server1]
            B --> D[Server2]
    </pre>

    And here is another:
    <pre class="mermaid">
            graph TD
            A[Client] -->|tcp_123| B
            B(Load Balancer)
            B -->|tcp_456| C[Server1]
            B -->|tcp_456| D[Server2]
    </pre>

    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
      mermaid.initialize({ startOnLoad: true });
    </script>
  </body>
</html>
```

</blockQuote>

...produced:

<p></p><img width="312" height="766" alt="image" src="https://github.com/user-attachments/assets/9066bc20-69a7-445b-94ea-3e070809880e" /><p></p>

**Comment by phil294** [comment]
To generate a static SVG/PNG, see https://github.com/mermaid-js/mermaid-cli
